File: gluecodium/src/test/resources/smoke/generic_types/output/android/com/example/smoke/EnumSets.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import java.util.Arrays;
import java.util.EnumSet;
import java.util.Set;
public final class EnumSets {
    public static final Set<GenericTypesWithCompoundTypes.SomeEnum> ENUM_SET_CONST = new EnumSet<>(Arrays.asList(GenericTypesWithCompoundTypes.SomeEnum.FOO, GenericTypesWithCompoundTypes.SomeEnum.BAR));
    @NonNull
    public Set<GenericTypesWithCompoundTypes.SomeEnum> enumSetField;
    public EnumSets() {
        this.enumSetField = new EnumSet<>();
    }
    public EnumSets(@NonNull final Set<GenericTypesWithCompoundTypes.SomeEnum> enumSetField) {
        this.enumSetField = enumSetField;
    }
}

File: gluecodium/src/test/resources/smoke/generic_types/output/android-kotlin/com/example/smoke/EnumSets.kt
/*

 *
 */

@file:JvmName("EnumSetsExtensions")


package com.example.smoke

import java.util.EnumSet

class EnumSets {
    @JvmField var enumSetField: Set<@JvmSuppressWildcards GenericTypesWithCompoundTypes.SomeEnum>



    constructor() {
        this.enumSetField = EnumSet.noneOf(GenericTypesWithCompoundTypes.SomeEnum::class.java)
    }





    companion object {
        @JvmField final val ENUM_SET_CONST: Set<@JvmSuppressWildcards GenericTypesWithCompoundTypes.SomeEnum> = EnumSet.of(GenericTypesWithCompoundTypes.SomeEnum.FOO, GenericTypesWithCompoundTypes.SomeEnum.BAR)
    }
}

